activerecord/test/cases/associations/belongs_to_associations_test.rb

```diff
@@ -147,6 +168,12 @@ def test_type_mismatch
     assert_raise(ActiveRecord::AssociationTypeMismatch) { Account.find(1).firm = Project.find(1) }
   end
 
+  def test_type_mismatch_with_namespaced_class
+    Admin::User.belongs_to(:region)
+    assert_nil defined?(Region), "This test should be done with a only namespaced class"
+    assert_raise(ActiveRecord::AssociationTypeMismatch) { Admin::User.new(region: 'wrong value') }
+  end
```

Comment:
You should be able to define everything inline. The classes for example:

```ruby
  def test_type_mismatch_with_namespaced_class
    Admin.const_set "RegionalUser", (Class.new(Admin::User) {
                                       belongs_to(:region)
                                     })
    Admin.const_set "Region", Class.new(Admin::User)
    e = assert_raise(ActiveRecord::AssociationTypeMismatch) {
      Admin::RegionalUser.new(region: 'wrong value')
    }
    assert_match(/^Region\([^)]+\) expected, got String\([^)]+\)$/, e.message)
  ensure
    Admin.send :remove_const, "Region" if Admin.const_defined?("Region")
    Admin.send :remove_const, "RegionalUser" if Admin.const_defined?("RegionalUser")
  end
```

Some goes with the schema modifications that you currently do in `setup` and `teardown`. No need to do that for every test.